c10/core/Scalar.h

```diff
@@ -158,14 +198,68 @@ class C10_API Scalar {
     }
   }
 
+  Scalar(Scalar&& rhs) noexcept : tag(rhs.tag) {
+    moveFrom(std::move(rhs));
+  }
+
+  Scalar(const Scalar& rhs) : Scalar(rhs.v, rhs.tag) {}
+
+  Scalar(c10::SymInt si) : tag(Tag::HAS_si) {
+    v.si = si;
+  }
+
+  C10_ALWAYS_INLINE void moveFrom(Scalar&& rhs) noexcept {
+    if (rhs.tag == Tag::HAS_si) {
+      new (&v.si) c10::SymInt(rhs.v.si);
```

Comment:
We should be able to just do
```
v.si = std::move(rhs.v.si)
```
It's a little bit scary you need to call `release_` on SymInt manually

?